build_tools/bazel/iree_bytecode_module.bzl

```
@@ -34,23 +35,34 @@ def iree_bytecode_module(
         linker_tool: the linker to use.
             Defaults to the lld from the llvm-project directory.
         c_identifier: Optional. Enables embedding the module as C data.
+        static_lib_path: When set, the module is compiled into a LLVM static
+            library with the specified library path.
         deps: Optional. Dependencies to add to the generated library.
         **kwargs: any additional attributes to pass to the underlying rules.
     """
 
     if not module_name:
         module_name = "%s.vmfb" % (name)
 
+    out_files = [module_name]
+    module_flags = ["--output-format=vm-bytecode"]
+    if static_lib_path:
+        static_header_path = static_lib_path.replace(".o", ".h")
+        out_files.extend([static_lib_path, static_header_path])
+        module_flags.extend([
+            "--iree-llvm-link-embedded=false",
+            "--iree-llvm-link-static",
+            "--iree-llvm-static-library-output-path=$(location %s)" % (static_lib_path),
+        ])
+
     native.genrule(
         name = name,
         srcs = [src],
-        outs = [
-            module_name,
-        ],
+        outs = out_files,
         cmd = " && ".join([
             " ".join([
                 "$(location %s)" % (compile_tool),
-                " ".join(["--output-format=vm-bytecode"] + flags),
+                " ".join(module_flags + flags),
```

Continuing from above

```suggestion
                " ".join(flags),
```

---

Done. Thanks for the suggestion.